### PR TITLE
adding preview/compile updates

### PR DESCRIPTION
**Pull request by mirnawong1**
adding preview / compile updates from @StefanDBTLabs 's  loom. this includes being able to preview/compile selected code (saved or unsaved)

for v1.6 and higher, preview defaults to --select model_name

### CODE REVIEW

File: website/docs/docs/cloud/dbt-cloud-ide/develop-in-the-cloud.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ To stay informed on IDE updates, read [dbt Cloud IDE release notes](/tags/ide),
...
 | **Project documentation** | Generate and view your [project documentation](/docs/collaborate/build-and-view-your-docs) for your dbt project in real-time. You can inspect and verify what your project's documentation will look like before you deploy your changes to production. |
-| **Preview and Compile button** | You can run your code against your data platform by clicking the [**Preview**](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section). Use the [**Compile**](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section) button in the IDE to generate executable SQL, which occurs locally within dbt. |
+| **Preview and Compile button** | You can run your code or selected code against your data platform by clicking the [**Preview**](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section). Use the [**Compile**](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section) button in the IDE to generate executable SQL, which occurs locally within dbt. |
```
Comment (mirnawong1) [inline website/docs/docs/cloud/dbt-cloud-ide/develop-in-the-cloud.md]:
```suggestion
| **Preview and Compile button** | You can run your code or selected code against your data platform by clicking the [**Preview**](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section) button. <br /> Use the [**Compile**](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section) button in the IDE to generate executable SQL, which occurs locally within dbt. |
```

File: website/docs/docs/cloud/dbt-cloud-ide/develop-in-the-cloud.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ To stay informed on IDE updates, read [dbt Cloud IDE release notes](/tags/ide),
...
 | **Project documentation** | Generate and view your [project documentation](/docs/collaborate/build-and-view-your-docs) for your dbt project in real-time. You can inspect and verify what your project's documentation will look like before you deploy your changes to production. |
-| **Preview and Compile button** | You can run your code against your data platform by clicking the [**Preview**](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section). Use the [**Compile**](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section) button in the IDE to generate executable SQL, which occurs locally within dbt. |
+| **Preview and Compile button** | You can run your code or selected code against your data platform by clicking the [**Preview**](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section) button. <br /> Use the [**Compile**](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section) button in the IDE to generate executable SQL, which occurs locally within dbt. |
```
Comment (jtcohen6) [inline website/docs/docs/cloud/dbt-cloud-ide/develop-in-the-cloud.md]: This isn't quite right - the change isn't around letting users highlight a subset of code and preview just that selected subset. The change is that you can now preview a model _using the actual context of that model_, same as you would get when building it.

So this is more like: "You can compile or preview a snippet of dbt code, or one of your dbt models after editing and saving."

File: website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md [hunk 3]
```diff
@@ -95,7 +95,7 @@ The dbt Cloud IDE returns default row limits, however, you can also specify the
 </ul>
 </details>
 
-2. **Compile button &mdash;** The Compile button compiles the SQL code from the active File Editor, irrespective of its save status, and outputs it to the Compiled Code tab.
+2. **Compile button &mdash;** The **Compile** button compiles the saved or unsaved SQL code from the active file editor, and outputs it to the **Compiled Code** tab.  Starting from dbt v1.6 or higher, you can compile a selected portion of saved or unsaved code by highlighting it before clicking the **Compile** button.
```
Comment (jtcohen6) [inline website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md]: As above - this isn't quite right. This suggestion isn't perfect wording, but gets closer to what the substantive change is.

```suggestion
2. **Compile button &mdash;** The **Compile** button compiles the saved or unsaved SQL code from the active file editor, and outputs it to the **Compiled Code** tab. Starting from dbt v1.6 or higher, after saving changes to a model, you can compile its code with the context of that model — same as you would get while building it — which includes context variables like `{{ this }}` or `{{ is_incremental() }}`.
```

**Review by jtcohen6 — COMMENTED** [review]
Thanks @mirnawong1!

File: website/docs/docs/cloud/dbt-cloud-ide/develop-in-the-cloud.md (hunk 2 above)
Comment (mirnawong1) [reply website/docs/docs/cloud/dbt-cloud-ide/develop-in-the-cloud.md]: love this, thank you!

Comment (mirnawong1) [inline website/docs/docs/cloud/dbt-cloud-ide/develop-in-the-cloud.md]:
```suggestion
| **Preview and Compile button** | You can [compile or preview](/docs/cloud/dbt-cloud-ide/ide-user-interface#console-section) code, a snippet of dbt code, or one of your dbt models after editing and saving. |
```

File: website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md [hunk 4]
```diff
@@ -95,7 +95,7 @@ The dbt Cloud IDE returns default row limits, however, you can also specify the
 </ul>
 </details>
 
-2. **Compile button &mdash;** The Compile button compiles the SQL code from the active File Editor, irrespective of its save status, and outputs it to the Compiled Code tab.
+2. **Compile button &mdash;** The **Compile** button compiles the saved or unsaved SQL code from the active file editor, and outputs it to the **Compiled Code** tab. Starting from dbt v1.6 or higher, after saving changes to a model, you can compile its code with the context of that model — same as you would get while building it — which includes context variables like `{{ this }}` or `{{ is_incremental() }}`.
```
Comment (mirnawong1) [inline website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md]:
```suggestion
2. **Compile button &mdash;** The **Compile** button compiles the saved or unsaved SQL code and displays it in the **Compiled Code** tab. From dbt version 1.6 onwards, when you save changes to a model, you can compile its code with the model's specific context. This context is similar to what you'd have when building the model and involves useful context variables like `{{ this }} `or `{{ is_incremental() }}`.


Starting from dbt v1.6 or higher, after saving changes to a model, you can compile its code with the context of that model — same as you would get while building it — which includes context variables like `{{ this }}` or `{{ is_incremental() }}`.
```

File: website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md [hunk 5]
```diff
@@ -83,7 +83,7 @@ The IDE features some delightful tools and layouts to make it easier for you to
 The console section, located below the File editor, includes various console tabs and buttons to help you with tasks such as previewing, compiling, building, and viewing the <Term id="dag" />. Refer to the following sub-bullets for more details on the console tabs and buttons.
 <Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-console-overview.jpg" width="100%" title="The Console section is located below the File editor and has various tabs and buttons to help execute tasks"/>
 
-1. **Preview button &mdash;** When you click on the Preview button, it runs the SQL in the active file editor regardless of whether you have saved it or not and sends the results to the Results console tab. 
+1. **Preview button &mdash;** When you click on the Preview button, it runs the SQL in the active file editor regardless of whether you have saved it or not and sends the results to the **Results** console tab. Starting from dbt v1.6 or higher, you can preview a selected portion of saved or unsaved code by highlighting it before clicking the **Preview** button.
```
Comment (mirnawong1) [inline website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md]:
```suggestion
1. **Preview button &mdash;** When you click on the Preview button, it runs the SQL in the active file editor regardless of whether you have saved it or not and sends the results to the **Results** console tab. From dbt v1.6 or higher, you can preview a selected portion of saved or unsaved code by highlighting it and then clicking the **Preview** button.
```

File: website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md [hunk 6]
```diff
@@ -83,7 +83,7 @@ The IDE features some delightful tools and layouts to make it easier for you to
 The console section, located below the File editor, includes various console tabs and buttons to help you with tasks such as previewing, compiling, building, and viewing the <Term id="dag" />. Refer to the following sub-bullets for more details on the console tabs and buttons.
 <Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-console-overview.jpg" width="100%" title="The Console section is located below the File editor and has various tabs and buttons to help execute tasks"/>
 
-1. **Preview button &mdash;** When you click on the Preview button, it runs the SQL in the active file editor regardless of whether you have saved it or not and sends the results to the Results console tab. 
+1. **Preview button &mdash;** When you click on the Preview button, it runs the SQL in the active file editor regardless of whether you have saved it or not and sends the results to the **Results** console tab. From dbt v1.6 or higher, you can preview a selected portion of saved or unsaved code by highlighting it and then clicking the **Preview** button.
```
Comment (StefanDBTLabs) [inline website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md]: you can highlight and run just that code with any dbt version.

Comment (mirnawong1) [inline website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md]:
```suggestion
1. **Preview button &mdash;** When you click on the Preview button, it runs the SQL in the active file editor regardless of whether you have saved it or not and sends the results to the **Results** console tab. You can preview a selected portion of saved or unsaved code by highlighting it and then clicking the **Preview** button.
```

File: website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md [hunk 7]
```diff
@@ -95,7 +95,10 @@ The dbt Cloud IDE returns default row limits, however, you can also specify the
 </ul>
 </details>
 
-2. **Compile button &mdash;** The Compile button compiles the SQL code from the active File Editor, irrespective of its save status, and outputs it to the Compiled Code tab.
+2. **Compile button &mdash;** The **Compile** button compiles the saved or unsaved SQL code and displays it in the **Compiled Code** tab. From dbt version 1.6 onwards, when you save changes to a model, you can compile its code with the model's specific context. This context is similar to what you'd have when building the model and involves useful context variables like `{{ this }} `or `{{ is_incremental() }}`.
+
+
+Starting from dbt v1.6 or higher, after saving changes to a model, you can compile its code with the context of that model — same as you would get while building it — which includes context variables like `{{ this }}` or `{{ is_incremental() }}`.
```
Comment (StefanDBTLabs) [inline website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md]: yes!

File: website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md [hunk 8]
```diff
@@ -95,7 +95,10 @@ The dbt Cloud IDE returns default row limits, however, you can also specify the
 </ul>
 </details>
 
-2. **Compile button &mdash;** The Compile button compiles the SQL code from the active File Editor, irrespective of its save status, and outputs it to the Compiled Code tab.
+2. **Compile button &mdash;** The **Compile** button compiles the saved or unsaved SQL code and displays it in the **Compiled Code** tab. From dbt version 1.6 onwards, when you save changes to a model, you can compile its code with the model's specific context. This context is similar to what you'd have when building the model and involves useful context variables like `{{ this }} `or `{{ is_incremental() }}`.
```
Comment (mirnawong1) [inline website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md]:
```suggestion
2. **Compile button &mdash;** The **Compile** button compiles the saved or unsaved SQL code and displays it in the **Compiled Code** tab.
```

**Review by StefanDBTLabs — APPROVED** [review]

File: website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md (hunk 7 above)
Comment (mirnawong1) [inline website/docs/docs/cloud/dbt-cloud-ide/ide-user-interface.md]:
```suggestion
Starting from dbt v1.6 or higher, when you save changes to a model, you can compile its code with the model's specific context. This context is similar to what you'd have when building the model and involves useful context variables like `{{ this }} `or `{{ is_incremental() }}`.
```

**Pull request merged**